runelite-client/src/main/java/net/runelite/client/util/WorldMapView.java

```diff
@@ -0,0 +1,210 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.util;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.RenderingHints;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import javax.imageio.ImageIO;
+import javax.inject.Inject;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.coords.WorldPoint;
+
+@Slf4j
+public class WorldMapView
+{
+	private static LoadingCache<Integer, BufferedImage> regions;
+
+	private static final int REGION_GRAPHIC_SIZE = 256;
+	private static final int REGION_TILE_SIZE = 64;
+
+	@Getter
+	private WorldPoint center;
+
+	@Getter
+	private float zoom;
+
+	@Getter
+	private int graphicsWidth;
+
+	@Getter
+	private int graphicsHeight;
+
+
+	public void setCenter(WorldPoint center)
+	{
+		this.center = center;
+		updateCalculations();
+	}
+
+	public void setZoom(float zoom)
+	{
+		this.zoom = zoom;
+		updateCalculations();
+	}
+
+	public void setGraphicsWidth(int graphicsWidth)
+	{
+		this.graphicsWidth = graphicsWidth;
+		updateCalculations();
+	}
+
+	public void setGraphicsHeight(int graphicsHeight)
+	{
+		this.graphicsHeight = graphicsHeight;
+		updateCalculations();
+	}
+
+	private int regionSizeScaled;
+	private Point tileOffset;
+	@Getter
+	private float pixelsPerTile;
+	private Point graphicalOffset;
+	private Point graphicalCenter;
+
+	private void updateCalculations()
+	{
+		//Calculate region graphical size.
+		regionSizeScaled = (int) (REGION_GRAPHIC_SIZE * zoom);
+		//How far from top left of region this tile is
+		tileOffset = new Point(center.getX() % REGION_TILE_SIZE, center.getY() % REGION_TILE_SIZE);
+		//Graphical size of tileOffset
+		pixelsPerTile = regionSizeScaled / (float)REGION_TILE_SIZE;
+		graphicalOffset = new Point((int)(tileOffset.x * pixelsPerTile), (int)(tileOffset.y * pixelsPerTile));
+		graphicalCenter = new Point(graphicsWidth / 2, graphicsHeight / 2);
+	}
+
+
+	@Inject
+	public WorldMapView()
+	{
+		if (regions == null)
+		{
+			regions = CacheBuilder.newBuilder()
+				.maximumSize(1685)
```

Comment:
Why do you need so big cache? You would be showing max like 9 regions at once and all older ones should be invalidated probably.

Reply:
There are 1685 map tiles, it is better to cache them than to have to hit the disk continually if you ever try to draw all of them. In the future there is the possibility of zoom levels etc but for now this seems like the safest solution.

Reply:
With a quick test I'm not actually seeing any difference when I change the cache size, so I will lower it.